### ISSUE TITLE
Ability to pin third-party toots

### ISSUE THREAD

**Issue by koyuawsmbrtn** [tootsuite/mastodon#12443]
### Pitch

Someone wrote something I would totally pin onto my profile, but there is no option to do that.

### Motivation

That makes me sad...and maybe other people

**Comment by Cassolotl** [comment]
Gargron on https://github.com/tootsuite/mastodon/issues/5182#issuecomment-333671271 in October 2017:

>You can't pin toots from other accounts and you won't be able to because it opens up various forms of abuse. Permanently displaying someone else's status on your profile is something that should require consent, and I don't believe it's worthwhile to build out the functionality for that. Sorry.

He might have changed his mind since then though, it's been a couple of years!

**Comment by koyuawsmbrtn** [comment]
I mean we could implement a way to forbid someone to pin a status. Deleting a status (or sending a delete request, because ActivityPub) should also trigger unpinning like the same behaviour on your own account.

**Comment by trwnh** [comment]
The same consent argument applies to featured profiles like it does featured posts, and iirc it's handled there by only allowing you to feature people you are following.

**Comment by MirceaKitsune** [comment]
Yes, I'd go for the same system personally: You should be allowed to pin someone else's toots on your profile, but only as long as you're following them.

**Comment by ffffluffy** [comment]
I guess we should at least have an option to opt-out. I personnally don't want anyone to be able to pin my toots (even the public ones) to their profile.

**Comment by trwnh** [comment]
>I personnally don't want anyone to be able to pin my toots (even the public ones) to their profile.

functionally it is similar to boosting a public post, and you could even pin an Announce activity (which is technically still "your" status, even if it's boosting someone else). there's no way to control what happens with public posts.

**Comment by nightpool** [comment]
to be clear: you can not, currently, pin an announce activity and I don't
think mastodon should allow you to do so.
